### ISSUE TITLE
Phase 2 Block 1 compatibility placeholder

### ISSUE THREAD

**Issue by mrmin123** [perryhuynh/kcauto#210]
Kancolle Phase 2 Block 1 is going live on 8/17 1800 JST. This update will break kcauto. Due to timezone differences (maint. ends 2 AM, my time) and work obligations, I will not be able to update kcauto for at least 12 hours after maintenance ends. I will be working on updating kcauto ASAP, but until then, please be patient.

Due to the potential scope of the work ahead (400+ images, 30+ map JSON files, unknown number of logic changes), I will be grouping and prioritizing the work in the following order:

1. Expeditions (and related)
2. PvP (and related)
3. Quests (and related)
4. Combat (and related)

Other new optimizations, features, and targets of opportunity will be tackled on an as-needed or convenience basis. Thank you.

**Comment by Danielosama** [comment]
Lot of work for you ahead man, good luck and thanks a lot!
If you need any help, say it. Don't know how I could help but who knows.

**Comment by n0k0m3** [comment]
Just played for 10 minutes. So far a lot of things changed (especially sorties maps) that can affects the script (also there're some improvements) [Will update with more pngs later]:
- Screen size upped to 1280x720. [It's huge](https://i.imgur.com/Kp2a0uz.jpg). (and we need a LOT of finetuning everywhere that needs mouse rejigger)
- [Expedition world icons](https://i.imgur.com/xrMy53G.jpg)
- Sortie maps (fuck this, too much work, gonna wait for wikia miners). HOWEVER: 1-6, 3-5 and world 6 stays the same (according to patchnote on reddit). [World icons changed as well](https://i.imgur.com/XQXGQUt.jpg).
- Ship sorting is now much better (easier to reach the SS list for ShipSwitcher)
  - Small [blue](https://i.imgur.com/svONj0k.png)/[red ](https://i.imgur.com/3X2ZWEE.png)button shows English Ship Classification (BB,CA,...) (this is for gameplay)
  - [Triangle](https://i.imgur.com/ZE9AAPu.png) hide/unhide all ship classes
  - You can [hide/show ship class](https://i.imgur.com/IEElZyk.png) that you desire

Seems like all other assests remain unchanged (idk about sortie though, haven't mess with even 1-1 yet), even Quest (thanks tanaka)

Edit: holy f Igirisu: https://www.reddit.com/r/kancolle/comments/981n9e/media_there_is_an_english_interface_now/

**Comment by yjjieqc** [comment]
Thanks for working on this.

**Comment by stackhanovets** [comment]
Dear God, just look at this:

![skins](https://user-images.githubusercontent.com/16455270/44283548-6a16a400-a267-11e8-98e2-156ee8d30b65.png)

There are five (5) dashboard skins! I can't say the game staff is informed about this tool but it was very clever.

So where do we start? Semi-english (bottom)? Inherited (top)? Else?

upd: 
There are still so much untranslated elements. It seems using anything besides the inherited Flash UI is not a good idea.

**Comment by stackhanovets** [comment]
Ok then, here comes some kind of manual hotfix from my side. Maybe it would help for somebody. **Please note - I'm using the generic version of kancolle-auto**

[screenshots.zip](https://github.com/mrmin123/kcauto/files/2299027/screenshots.zip)

**Comment by ForsakenRei** [comment]
Thanks! I think the most fortunate part is that we don't need to do all the screenshots but can get most of the resources from html5 directly. It does take time since there are lots of different parts in this game. Just keep your own peace.

P.S. I'm interested in Python and learned a little by myself. Then I'm wondering are there any possibility to make kcauto send messages to kancolle using win32api (libraries like pywinauto?) at background without moving the actual cursor therefore we can do our own staff when kcauto is dealing with the game. Sadly I have only 1 PC.

**Comment by perryhuynh** [comment]
Using win32api would limit users to only using Windows for this, have you looked into using a VM?

**Comment by stackhanovets** [comment]
@ForsakenRei 

Using Sikuli at first glance looks less detectable than using software-testing frameworks for web applications like Selenium. Good solution is another automation tool based on JavaScript, not Jython, you know.

**Comment by ForsakenRei** [comment]
@stackhanovets Of course. I just think Python can do that since I'm learning something about opencv and suddenly found it may work similar as sikuli if I can programming some pywinauto scripts for it. They are different approaches to do the same thing but my choice might be much more complicated. After all it is just a thought during my study and not a request, maybe I can use kcauto as a reference in the future and try to do something myself.

**Comment by mrmin123** [comment]
[4.0.0](https://github.com/mrmin123/kcauto/releases/tag/4.0.0) brings limited compatibility with the KC update.

If you are on Chrome and are getting constant `FindFailed` errors, please open `chrome://flags/#force-color-profile` in Chrome, and set the 'Force color profile' flag to 'sRGB', then restart your browser.

**Comment by mrmin123** [comment]
@ForsakenRei @stackhanovets for a Python solution, I'm beginning to think something using [pyautogui](https://github.com/asweigart/pyautogui) could work. This does mean, however, manually binding OpenCV and TesseractOCR, which is really what Sikuli abstracts out. I personally think that a pure JS solution running within Chrome (like KC3) is the way to go, since you have direct access to the API payloads (and potentially the DOM), but I have my reservations on its ability to mock mouse clicks that will actually be recognized as mouse clicks by whatever anti-bot measure KC now has.

**Comment by ForsakenRei** [comment]
@mrmin123 Thanks for the reply. 
To my knowledge there are some of people using scripts from QuickMacro or other hacks to send background mouse clicks with random biases each time. So far they are good and only get a code 201 error if battle at 2-3 too much in a day for example. But no ban for about 2 years while KC said anti-bot is reinforced every time. I do know your concern about that so VM maybe the best way now.

I found a thread on stackoverflow about [pywinauto](https://github.com/pywinauto/pywinauto) which shows background click can be done but still you might need manually bind OpenCV and OCR to play KC (and only in Win) LuL.

**Comment by stackhanovets** [comment]
Just in case, JUnit as unit test automation + PhantomJS as automation framework + Selenium as functional testing automation tool is very poplar bundle for CD/CI DevOps workflow. Reasons: speed, reliability (or predictability IYKWIM) and community.

**Comment by stackhanovets** [comment]
OK, I've got the loop:

```
[] Sortieing fleet 2 to expedition 2
[log] CLICK on L(202,773)@S(0)[0,0 res] (531 msec)
[log] CLICK on L(202,777)@S(0)[0,0 res] (535 msec)
[log] CLICK on L(195,772)@S(0)[0,0 res] (529 msec)
[log] CLICK on L(205,770)@S(0)[0,0 res] (529 msec)
[log] CLICK on L(213,771)@S(0)[0,0 res] (530 msec)
[log] CLICK on L(214,775)@S(0)[0,0 res] (529 msec)
[log] CLICK on L(185,771)@S(0)[0,0 res] (533 msec)
[log] CLICK on L(206,772)@S(0)[0,0 res] (532 msec)
[log] CLICK on L(214,781)@S(0)[0,0 res] (539 msec)
[log] CLICK on L(191,769)@S(0)[0,0 res] (531 msec)
[log] CLICK on L(213,772)@S(0)[0,0 res] (531 msec)
[log] CLICK on L(209,778)@S(0)[0,0 res] (532 msec)
[log] CLICK on L(219,774)@S(0)[0,0 res] (536 msec)

```

It stuck on `e_world_1.png`. `chrome://flags/#force-color-profile -> sRGB` did not help.

**Comment by mrmin123** [comment]
@stackhanovets @perryhuynh had the same issue. He had to reset the custom DPI setting on his desktop environment.

**Comment by perryhuynh** [comment]
Not reset it rather but disable the DPI scaling for fonts altogether, it seemed to have been enabled by default when installed the OS but never affected it before most likely because it was running on Flash Player

**Comment by etcet54** [comment]
similar to stackhavonets, I experienced loop but when fleet 4 going to expedition 38, it keep clicking the expedition world icon.
Tried changing the color profile to sRGB and have checked no scaling used on windows display setting(100%), and chrome properties no scaling box checkmarked, issue still exist.

I changed to expedition 11 and it worked, so for now I couldn't send it to world 6 where expedition 37 and 38 located.

**Comment by stackhanovets** [comment]
No, there are all default setup on my client. Hell, is there a difference? (mine vs provided):

![ensei_name_02](https://user-images.githubusercontent.com/16455270/44305679-5877ee00-a386-11e8-9ec7-25da163f3ba1.png)
![expedition_2](https://user-images.githubusercontent.com/16455270/44305671-367e6b80-a386-11e8-9622-82887bf77eed.png)

![ensei_area_01](https://user-images.githubusercontent.com/16455270/44305683-704f7200-a386-11e8-90db-720c24818b77.png)
![e_world_1](https://user-images.githubusercontent.com/16455270/44305681-69c0fa80-a386-11e8-995e-b38f8bdde749.png)

I've also tried to change the alias `/usr/bin/google-chrome-stable --high-dpi-support=1 --force-device-scale-factor=1 %U ` from [the topic](https://superuser.com/questions/1189725/disable-chrome-scaling) without effect.

**Comment by mrmin123** [comment]
There's definitely an issue caused by the rendering of fonts in HTML5 canvas across browsers/OSes. That's the same exact issue I saw w/ Perry's game on Unix vs the assets generated off my machine (Windows). Check the difference between the two images in photoshop and you get this:
![image](https://user-images.githubusercontent.com/4326866/44305883-761e6680-a338-11e8-82c6-9e5c2d2ac20a.png)
(black being complete match). The two images are aligned on the `E` symbol, which is a graphic and not a rendered font, so there's no differences there.

The DPI setting worked for Perry, so you might have to play around with the DPI settings on your machine, or I'll have to find an alternative implementation for identifying Expeditions.

Some additional reading: [Font Smooth Explained](https://szafranek.net/blog/2009/02/22/font-smoothing-explained/), [Subpixel anti-aliased text on HTML5's canvas element](https://stackoverflow.com/questions/4550926/subpixel-anti-aliased-text-on-html5s-canvas-element)

**Comment by mrmin123** [comment]
A viable workaround might be to lower the similarity threshold for the expedition lines.

@stackhanovets could you try modifying `L158` in `expedition.py`:
```python
        while not self.kc_region.exists(expedition_img):
```
to
```python
        while not self.kc_region.exists(Pattern(expedition_img).similar(0.7)):
```

And maybe experiment with the value in `similar()` to see if you can get it to match.

**Comment by etcet54** [comment]
@mrmin123 

hey, just wanna update, after I see your comment about image align issue, I tried to overwrite the expedition_38 image provided with my own screenshot, and it immediately worked now, thank you.

**Comment by stackhanovets** [comment]
@mrmin123 ,

```
[2018-08-19 TIME] Initializing config module
[2018-08-19 TIME] Reading config
[2018-08-19 TIME] Validating config
[2018-08-19 TIME] Starting kcauto!
[log] App.focus:  [pid:Chrome]
[2018-08-19 TIME] Quests reset.
[log] App.focus:  [pid:Chrome]
[2018-08-19 TIME] At side menu
[2018-08-19 TIME] Going home.
[log] CLICK on L(42,490)@S(0)[0,0 XxY] (528 msec)
[2018-08-19 TIME] FindFailed: screen_equipment_ready.png: (140x52) in R[607,472 600x360]@S(0) E:Y, T:1.0
  Line 2759, in file Region.java

[2018-08-19 TIME] ** FindFailed error occurred; attempting basic recovery. **
[log] App.focus:  [pid:Chrome]
[log]  TYPE "#ESC."
[log]  TYPE " "
[2018-08-19 TIME] Recovery successful.
[2018-08-19 TIME] At side menu
[2018-08-19 TIME] Going home.
[log] CLICK on L(47,340)@S(0)[0,0 XxY] (529 msec)
[log] CLICK on L(124,512)@S(0)[0,0 XxY] (528 msec)
[log] App.focus:  [pid:Chrome]
[2018-08-19 TIME] At home
[2018-08-19 TIME] At home
[2018-08-19 TIME] At home
[2018-08-19 TIME] Navigating to expedition screen.
[log] CLICK on L(302,479)@S(0)[0,0 XxY] (529 msec)
[log] CLICK on L(1041,474)@S(0)[0,0 XxY] (539 msec)
[2018-08-19 TIME] Sortieing fleet 2 to expedition 2
[2018-08-19 TIME] FindFailed: expedition_timer.png: (200x30) in R[7,112 1200x720]@S(0) E:Y, T:1.0
  Line 2759, in file Region.java

[2018-08-19 TIME] ** FindFailed error occurred; attempting basic recovery. **
[log] App.focus:  [pid:Chrome]
[log]  TYPE "#ESC."
[log]  TYPE " "
[2018-08-19 TIME] Recovery successful.
[2018-08-19 TIME] At side menu
[2018-08-19 TIME] Going home.
[log] CLICK on L(117,513)@S(0)[0,0 XxY] (528 msec)
[log] App.focus:  [pid:Chrome]
[2018-08-19 TIME] At home
[2018-08-19 TIME] At home
[2018-08-19 TIME] At home
[2018-08-19 TIME] Navigating to expedition screen.
[log] CLICK on L(305,513)@S(0)[0,0 XxY] (526 msec)
[log] CLICK on L(1022,462)@S(0)[0,0 XxY] (553 msec)
[2018-08-19 TIME] Sortieing fleet 2 to expedition 2
[2018-08-19 TIME] FindFailed: expedition_timer.png: (200x30) in R[7,112 1200x720]@S(0) E:Y, T:1.0
  Line 2759, in file Region.java

[2018-08-19 TIME] ** FindFailed error occurred; attempting basic recovery. **
[log] App.focus:  [pid:Chrome]
[log]  TYPE "#ESC."
[log]  TYPE " "
[2018-08-19 TIME] Recovery successful.
[2018-08-19 TIME] At side menu
[2018-08-19 TIME] Going home.
[log] CLICK on L(42,582)@S(0)[0,0 XxY] (527 msec)
[log] CLICK on L(124,497)@S(0)[0,0 XxY] (529 msec)
[log] App.focus:  [pid:Chrome]
[2018-08-19 TIME] At home
[2018-08-19 TIME] At home
[2018-08-19 TIME] At home
[2018-08-19 TIME] Navigating to expedition screen.
[log] CLICK on L(300,503)@S(0)[0,0 XxY] (528 msec)
[log] CLICK on L(986,451)@S(0)[0,0 XxY] (531 msec)
[2018-08-19 TIME] Sortieing fleet 2 to expedition 2
[2018-08-19 TIME] FindFailed: expedition_timer.png: (200x30) in R[7,112 1200x720]@S(0) E:Y, T:1.0
  Line 2759, in file Region.java

[2018-08-19 TIME] ** FindFailed error occurred; attempting basic recovery. **
[log] App.focus:  [pid:Chrome]
[log]  TYPE "#ESC."
[log]  TYPE " "
[2018-08-19 TIME] Recovery successful.
[2018-08-19 TIME] At side menu
[2018-08-19 TIME] Going home.
[log] CLICK on L(118,500)@S(0)[0,0 XxY] (528 msec)
[log] App.focus:  [pid:Chrome]
[2018-08-19 TIME] At home
[2018-08-19 TIME] At home
[2018-08-19 TIME] At home
[2018-08-19 TIME] Navigating to expedition screen.
[log] CLICK on L(278,506)@S(0)[0,0 XxY] (527 msec)
[log] CLICK on L(981,460)@S(0)[0,0 XxY] (536 msec)
[2018-08-19 TIME] Sortieing fleet 2 to expedition 2
[2018-08-19 TIME] FindFailed: expedition_timer.png: (200x30) in R[7,112 1200x720]@S(0) E:Y, T:1.0
  Line 2759, in file Region.java

[2018-08-19 TIME] ** FindFailed error occurred; attempting basic recovery. **
[log] App.focus:  [pid:Chrome]
[log]  TYPE "#ESC."
[log]  TYPE " "
[2018-08-19 TIME] Recovery successful.
[2018-08-19 TIME] At side menu
[2018-08-19 TIME] Going home.
[log] CLICK on L(116,451)@S(0)[0,0 XxY] (531 msec)
[log] App.focus:  [pid:Chrome]
[2018-08-19 TIME] At home
[2018-08-19 TIME] At home
[2018-08-19 TIME] At home
[2018-08-19 TIME] Navigating to expedition screen.
[log] CLICK on L(325,518)@S(0)[0,0 XxY] (529 msec)
[log] CLICK on L(1029,409)@S(0)[0,0 XxY] (530 msec)
[2018-08-19 TIME] Sortieing fleet 2 to expedition 2
[2018-08-19 TIME] FindFailed: expedition_timer.png: (200x30) in R[7,112 1200x720]@S(0) E:Y, T:1.0
  Line 2759, in file Region.java

[2018-08-19 TIME] ** FindFailed error occurred; attempting basic recovery. **
[log] App.focus:  [pid:Chrome]
[log]  TYPE "#ESC."
[log]  TYPE " "
[2018-08-19 TIME] Recovery successful.

```
upd:
No crashes after rolling back. Except the loop of course.

upd:
@mrmin123  which OS did you use for screen capture?

**Comment by mrmin123** [comment]
@stackhanovets the looping `FindFailed` seems like a color space error to me. Did you force sRGB on chrome? `chrome://flags/#force-color-profile`, 'Force color profile' flag to 'sRGB'

Assets were generated on Win7, Chrome w/ color profile set to sRGB.

**Comment by stackhanovets** [comment]
Yes, I did.

![enabled](https://user-images.githubusercontent.com/16455270/44306481-79493f00-a398-11e8-8dcb-67deaa709b29.png)

OK then, I'm going into the botnet way again. I wish I could disable M$ Update services 😂

**Comment by perryhuynh** [comment]
What desktop environment are you using?

**Comment by stackhanovets** [comment]
@perryhuynh XFCE and KDE were tested.

**Comment by perryhuynh** [comment]
And you disabled this in XFCE and restarted? Its enabled by default I believe
![image](https://user-images.githubusercontent.com/9361780/44306596-1e3b2c00-a347-11e8-85ed-ea998c64f768.png)

**Comment by stackhanovets** [comment]
**tl:dr It seems we are now bound with Win7 - because of fonts.😂👌💯 
Hope @mrmin123 won't migrate at near future. 😅**

So I've just installed Window7 x64 Ultimate en-US within another VM instance. I'm feeling lucky I've got a spare SSD. As I expected, the expedition module was working just fine, however, the script has crashed:

```
[error] RunTimeINIT: *** BE AWARE: Running on Java 8+ *** Please report problems ***
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.python.core.PySystemState to method java.io.Console.encoding()
WARNING: Please consider reporting this to the maintainers of org.python.core.PySystemState
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

[94m[2018-08-19 TIME] Initializing config module[0m
[94m[2018-08-19 TIME] Reading config[0m
[94m[2018-08-19 TIME] Validating config[0m
[94m[2018-08-19 TIME] Starting kcauto![0m
[log] App.focus:  [3848:Chrome]
[94m[2018-08-19 TIME] Quests reset.[0m
[log] App.focus:  [3848:Chrome]
[94m[2018-08-19 TIME] At side menu[0m
[94m[2018-08-19 TIME] Going home.[0m
[log] CLICK on L(40,573)@S(0)[0,0 RESOLUTION] (526 msec)
[log] CLICK on L(130,503)@S(0)[0,0 RESOLUTION] (528 msec)
[94m[2018-08-19 TIME] At home[0m
[94m[2018-08-19 TIME] At home[0m
[94m[2018-08-19 TIME] Navigating to quests screen.[0m
[log] CLICK on L(493,450)@S(0)[0,0 RESOLUTION] (531 msec)
[log] CLICK on L(136,346)@S(0)[0,0 RESOLUTION] (527 msec)
[log] CLICK on L(141,391)@S(0)[0,0 RESOLUTION] (536 msec)
[94m[2018-08-19 TIME] Activating quest d3.[0m
[94m[2018-08-19 TIME] Quest queue full.[0m
[log] CLICK on L(943,797)@S(0)[0,0 RESOLUTION] (526 msec)
[log] CLICK on L(98,436)@S(0)[0,0 RESOLUTION] (532 msec)
[log] CLICK on L(128,473)@S(0)[0,0 RESOLUTION] (538 msec)
[94m[2018-08-19 TIME] At top menu[0m
[94m[2018-08-19 TIME] Going home.[0m
[log] CLICK on L(35,782)@S(0)[0,0 RESOLUTION] (522 msec)
[log] App.focus:  [3848:Chrome]
[94m[2018-08-19 TIME] At home[0m
[94m[2018-08-19 TIME] At home[0m
[94m[2018-08-19 TIME] At home[0m
[94m[2018-08-19 TIME] Navigating to expedition screen.[0m
[log] CLICK on L(300,469)@S(0)[0,0 RESOLUTION] (523 msec)
[log] CLICK on L(1041,458)@S(0)[0,0 RESOLUTION] (521 msec)
[94m[2018-08-19 TIME] Sortieing fleet 2 to expedition 6[0m
[log] CLICK on L(511,577)@S(0)[0,0 RESOLUTION] (531 msec)
[log] CLICK on L(1078,743)@S(0)[0,0 RESOLUTION] (533 msec)
[93m[2018-08-19 TIME] Fleet needs resupply![0m
[log] CLICK on L(753,754)@S(0)[0,0 RESOLUTION] (523 msec)
[94m[2018-08-19 TIME] Resupplying with supply fairy.[0m
[log] CLICK on L(997,765)@S(0)[0,0 RESOLUTION] (538 msec)
[94m[2018-08-19 TIME] Fleet 2 sortied. Expected return time: 2018-08-19 TIME[0m
[94m[2018-08-19 TIME] Sortieing fleet 3 to expedition 5[0m
[log] CLICK on L(591,542)@S(0)[0,0 RESOLUTION] (524 msec)
[94m[2018-08-19 TIME] Got valid timer (TIME)![0m
[93m[2018-08-19 TIME] Expedition is already running. Return time: 2018-08-19 TIME[0m
[94m[2018-08-19 TIME] Sortieing fleet 4 to expedition 37[0m
[log] CLICK on L(515,762)@S(0)[0,0 RESOLUTION] (527 msec)
[log] CLICK on L(501,544)@S(0)[0,0 RESOLUTION] (525 msec)
[94m[2018-08-19 TIME] Got valid timer (TIME)![0m
[93m[2018-08-19 TIME] Expedition is already running. Return time: 2018-08-19 TIME[0m
[log] App.focus:  [3848:Chrome]
[94m[2018-08-19 TIME] At side menu[0m
[94m[2018-08-19 TIME] Going home.[0m
[log] CLICK on L(40,410)@S(0)[0,0 RESOLUTION] (521 msec)
[log] CLICK on L(129,489)@S(0)[0,0 RESOLUTION] (535 msec)
[94m[2018-08-19 TIME] At home[0m
[94m[2018-08-19 TIME] At home[0m
[94m[2018-08-19 TIME] Navigating to quests screen.[0m
[log] CLICK on L(157,348)@S(0)[0,0 RESOLUTION] (525 msec)
[log] CLICK on L(134,394)@S(0)[0,0 RESOLUTION] (536 msec)
[94m[2018-08-19 TIME] Activating quest c2.[0m
[94m[2018-08-19 TIME] Quest queue full.[0m
[94m[2018-08-19 TIME] Activating quest d3.[0m
[94m[2018-08-19 TIME] Quest queue full.[0m
[log] CLICK on L(947,791)@S(0)[0,0 RESOLUTION] (522 msec)
[log] CLICK on L(102,434)@S(0)[0,0 RESOLUTION] (538 msec)
[log] CLICK on L(118,480)@S(0)[0,0 RESOLUTION] (524 msec)
[94m[2018-08-19 TIME] At top menu[0m
[94m[2018-08-19 TIME] Going home.[0m
[log] CLICK on L(55,781)@S(0)[0,0 RESOLUTION] (524 msec)
[94m[2018-08-19 TIME] At home[0m
[94m[2018-08-19 TIME] At home[0m
[94m[2018-08-19 TIME] Navigating to fleetcomp screen.[0m
[log] CLICK on L(113,449)@S(0)[0,0 RESOLUTION] (524 msec)
[log] CLICK on L(48,318)@S(0)[0,0 RESOLUTION] (531 msec)
[log] CLICK on L(130,673)@S(0)[0,0 RESOLUTION] (532 msec)
[94m[2018-08-19 TIME] Switching to fleet preset 1.[0m
[91m[2018-08-19 TIME] Could not find fleet preset 1. Please check your config and fleet prests.[0m
[info] Exit code: 1

```
@perryhuynh I've tried to have a deal with anti-aliasing, some Chrome WebKit tricks, even hardware acceleration. Unfortunately, I was keeping receive errors. For now it looks like I used a half-measure. Maybe it was just some kind of bad luck in my case, however, I'd like to test this radical solution for a while.

**Comment by mrmin123** [comment]
Anything related to combat has not been ported over right now, including
fleet switcher. Disable that and combat until kcauto is patched.

**Comment by waicool20** [comment]
Resetting my dpi does not work for me, even if i can get the fonts to align up for the most part, the windows ones (asset) seem to be rendered slightly "taller", I'll be generating my own assets for now

**Comment by waicool20** [comment]
Since I couldn't get any combination of settings to work for me I just opted to modify the script/assets

I modified line 157 in expedition.py to 
`expedition_img = Pattern('expedition_{}.png'.format(fleet.expedition)).similar(0.75)`
then cropped all the existing assets to 380 length with imagemagick using the command
`mogrify -crop 380x30+0+0 expedition_*` in the expedition asset folder

**Comment by mrmin123** [comment]
[4.1.0](https://github.com/mrmin123/kcauto/releases/tag/4.1.0) ([related ticket](https://github.com/mrmin123/kcauto/issues/214)) brings further compatibility and a number of bugfixes, including better cross-browser/OS expedition support. Please read the changelog to see compatibility and supported UI options!

Please report additional issues here.

**Comment by stackhanovets** [comment]
LOG

```
[94m[2018-08-21 TIME] Initializing config module[0m
[94m[2018-08-21 TIME] Reading config[0m
[94m[2018-08-21 TIME] Validating config[0m
[94m[2018-08-21 TIME] Starting kcauto![0m
[log] App.focus:  [2640:Chrome]
[error] script [ C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli ] stopped with error at line --unknown--
[error] Error caused by: Traceback (most recent call last):
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\kcauto.py", line 53, in <module>
    kcauto.refresh_config()
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\kcauto.py", line 53, in <module>
    kcauto.refresh_config()
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\main.py", line 85, in refresh_config
    self._focus_kancolle()
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\main.py", line 192, in _focus_kancolle
    self.kc_region, self.regions = Util.focus_kc(self.config)
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\util.sikuli\util.py", line 321, in focus_kc
    kc = cls.focus_app(config)
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\util.sikuli\util.py", line 305, in focus_app
    kc.setAutoWaitTimeout(1)
AttributeError: 'NoneType' object has no attribute 'setAutoWaitTimeout'

<>

[log] CLICK on L(1050,596)@S(0)[0,0 1920x1080] (537 msec)
[94m[2018-08-21 TIME] Fleet 1 damage counts: 0 heavy / 0 moderate / 0 minor[0m
[94m[2018-08-21 TIME] Continuing sortie.[0m
[log] CLICK on L(483,447)@S(0)[0,0 1920x1080] (547 msec)
[94m[2018-08-21 TIME] Fleet at Node J[0m
[94m[2018-08-21 TIME] Formation specified for node J via map data.[0m
[94m[2018-08-21 TIME] Engaging the enemy in line abreast formation.[0m
[log] CLICK on L(1020,587)@S(0)[0,0 1920x1080] (597 msec)
[94m[2018-08-21 TIME] Fleet 1 damage counts: 0 heavy / 1 moderate / 1 minor[0m
[92m[2018-08-21 TIME] Sortie complete. Encountered 4 combat nodes (nodes A, D, F, J).[0m
[94m[2018-08-21 TIME] At home[0m
[94m[2018-08-21 TIME] At home[0m
[94m[2018-08-21 TIME] Navigating to repair screen.[0m
[log] CLICK on L(199,410)@S(0)[0,0 1920x1080] (589 msec)
[log] CLICK on L(100,580)@S(0)[0,0 1920x1080] (534 msec)
[94m[2018-08-21 TIME] Begin repairing fleets.[0m
[error] script [ C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli ] stopped with error at line --unknown--
[error] Error caused by: Traceback (most recent call last):
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\kcauto.py", line 68, in <module>
    kcauto.run_repair_cycle()
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\kcauto.py", line 68, in <module>
    kcauto.run_repair_cycle()
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\main.py", line 405, in run_repair_cycle
    self.modules['repair'].repair_fleets()
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\repair.sikuli\repair.py", line 77, in repair_fleets
    self._set_shiplist_counts()
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\repair.sikuli\repair.py", line 258, in _set_shiplist_counts
    self.ship_count, self.ship_page_count, self.ship_last_page_count = (
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\util.sikuli\util.py", line 228, in get_shiplist_counts
    ship_count = cls._get_ship_count(regions)
  File "C:\Users\USERNAME\Documents\Tools\kcauto\kcauto.sikuli\util.sikuli\util.py", line 255, in _get_ship_count
    number_read = int(number_read)
ValueError: invalid literal for int() with base 10: ''
```

**Comment by mrmin123** [comment]
@stackhanovets you're on UI 2, 4, or 5?

**Comment by stackhanovets** [comment]
@mrmin123 I'm using default inherited Flash KanColle UI v1.

**Comment by mrmin123** [comment]
Please read the changelog. It says which UIs are compatible.
